packages/utils/src/queryGenerators/createQueryString.ts

```diff
@@ -22,13 +22,30 @@ export const createQueryString = (
   prevKey: string = '',
 ): string => {
   const { fields = [] } = getTableByName(tablesList, tableName) || {};
-  const { deep = DEFAULT_DEPTH, withMeta = true, relationItemsCount, includeColumns } = queryObjectConfig;
+  const {
+    deep = DEFAULT_DEPTH,
+    withMeta = true,
+    relationItemsCount,
+    includeColumns,
+    permissions = {},
+  } = queryObjectConfig;
 
   let queryObject = '';
 
   fields
     .filter(field => {
       const isMeta = tableFieldSelectors.isMetaField(field);
+      const isRelation = tableFieldSelectors.isRelationField(field);
+
+      if (isRelation) {
+        const refTableName = tableFieldSelectors.getRelationTableName(field);
+
+        if (!R.pathOr(true, ['data', refTableName, 'permission', 'read', 'allow'], permissions)) {
+          return false;
```

Comment:
@zouxuoz Why don't you return just result of the `pathOr` function?

Reply:
@vorobeez  I rewrite this for more clarity, check updated code